sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -746,6 +784,271 @@ case class DescribeTableCommand(
   }
 }
 
+/**
+ * Command that looks like
+ * {{{
+ *   DESCRIBE [EXTENDED|FORMATTED] table_name partitionSpec? [AS JSON];
+ * }}}
+ */
+case class DescribeTableJsonCommand(
+   table: TableIdentifier,
+   partitionSpec: TablePartitionSpec,
+   isExtended: Boolean)
+  extends DescribeCommandBase {
+
+  override val output = DescribeCommandSchema.describeTableAttributes(true)
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val result = new ArrayBuffer[Row]
+    val catalog = sparkSession.sessionState.catalog
+
+    val schema = if (catalog.isTempView(table)) {
+      if (partitionSpec.nonEmpty) {
+        throw QueryCompilationErrors.descPartitionNotAllowedOnTempView(table.identifier)
+      }
+      catalog.getTempViewOrPermanentTableMetadata(table).schema
+    } else {
+      val metadata = catalog.getTableRawMetadata(table)
```

Comment:
We access the metadata again later: https://github.com/apache/spark/pull/49139/files#diff-632efff0ffceb2ff47b97f39570b4f2cff24401f132f559158acaa651b27af26R822

we should only access once.